projects/storefrontlib/cms-components/product/product-list/product-facet-navigation/facet-list/facet-list.component.ts

```diff
@@ -49,27 +60,46 @@ export class FacetListComponent {
   @Output() closeList = new EventEmitter();
 
   /** The list of all facet and values related to the products in the list */
-  facetList$: Observable<FacetList> = this.facetService.facetList$;
+  facetList$: Observable<FacetList>;
 
   iconTypes = ICON_TYPE;
 
   dialogFocusConfig: FocusConfig = {
     trap: true,
     block: true,
     focusOnEscape: true,
-    autofocus: 'cx-facet',
+    autofocus: 'cx-facet > button',
   };
 
   @HostListener('click') handleClick() {
     this.close();
   }
+  @Optional() focusService = inject(KeyboardFocusService, { optional: true });
+  @Optional() featureConfigService = inject(FeatureConfigService, {
+    optional: true,
+  });
 
   constructor(
     protected facetService: FacetService,
     protected elementRef: ElementRef,
     protected renderer: Renderer2
   ) {}
 
+  ngOnInit(): void {
+    this.facetList$ = this.facetService.facetList$.pipe(
```

Comment:
I'm still not really sure about reassigning this property. Perhaps we can tie it to a subscription instead?

Reply:
What do you mean by "reassigning"? This is the first tijme this property will be assigned a value(I've remove initialization in line 63. 
I didn't want to create a separate subscription for it since it seemed a bit redundant. Since we already have the subcription for the facetList$ observable using async pipe, I decided it will be slightly better to perform this side effect here instead of creating a separate subscription. What do you think?

Reply:
The reason for subscribing to the observable instead is to avoid the observable itself `facetList$` also being responsible for the `tap` whenever it is used by anything that may also use this component. Having the Observable be "dumb" could help us avoid `handleDialogFocus` being called more than once if the `facetList$` observable is used by any external component of service.

I think you might be right in saving performance by giving it the extra function and I think this would be fine in the main case. But I wonder if it could cause issues because of the reason above which could ultimately hinder performance or cause bugs. If we move this to a subscription instead in the component, I believe we would only call `handleDialogFocus` once on any `this.facetService.facetList$` changes instead of potentially more times if the component's property is observed.

This is my opinion though and I am happy to discuss this further :)

Reply:
You convinced me, I will move this to a separate subscription